### PR TITLE
[11.0][ADD] website_multi

### PR DESCRIPTION
**Pull request by tarteo**
This module makes it easier to manage multiple websites in one database.

Features
--------

* Adds configuration menu for websites
* Improves the list view of websites with extra fields
* Easy switch between websites
* Change the website of pages in the page properties dialog

The module still needs more features. For our customer the implemented features were adequate.

### CODE REVIEW

File: website_multi/menuitems.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Onestein
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <menuitem id="menu_configure_multi_website"
+        parent="website.menu_website_global_configuration"
+        action="action_configure_multi_website"/>
+
+    <menuitem id="menu_configure_menu_website"
```
Comment (simahawk) [inline website_multi/menuitems.xml:10]: same id same action?

File: website_multi/models/website.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2018 Onestein
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    # Replaces the web.base.url system parameter
```
Comment (simahawk) [inline website_multi/models/website.py]: can you explain?

File: website_multi/static/src/js/website_multi.editor.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            });
+
+            return this._super.apply(this, arguments);
+        }
+    });
+
+    var SwitchWebsiteMenu = WebsiteNavbarActionWidget.extend({
+        start: function () {
+            var $current_website = this.$('.current-website');
+            $current_website.html(location.hostname);
```
Comment (simahawk) [inline website_multi/static/src/js/website_multi.editor.js]: location coming from window?

File: website_multi/static/src/js/website_multi.editor.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+            var $dropdown_menu = this.$('.dropdown-menu');
+            rpc.query({
+                model: 'website',
+                method: 'search_read',
+                fields: ['domain']
+            }).then(function (websites) {
+                _.each(websites, function(website) {
+                    var url = location.port ? 'http://' + website.domain + ':' + location.port : 'http://' + website.domain;
```
Comment (simahawk) [inline website_multi/static/src/js/website_multi.editor.js]: why location info is used for each website?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_multi/models/website.py (hunk 2 above)
Comment (tarteo) [reply website_multi/models/website.py]: For example to use in generating the sitemap.xml

File: website_multi/static/src/js/website_multi.editor.js (hunk 3 above)
Comment (tarteo) [reply website_multi/static/src/js/website_multi.editor.js]: Yes.

File: website_multi/static/src/js/website_multi.editor.js (hunk 4 above)
Comment (tarteo) [reply website_multi/static/src/js/website_multi.editor.js]: I'll change it. I can use base_url here it must been a leftover before I had the field.

File: website_multi/menuitems.xml (hunk 1 above)
Comment (tarteo) [reply website_multi/menuitems.xml:10]: Can you explain this comment a little more, please?

File: website_multi/models/website.py (hunk 2 above)
Comment (simahawk) [reply website_multi/models/website.py]: but why replace? Shouldn't you reuse it instead or use it as a default?
Is it just for the switcher?

File: website_multi/menuitems.xml (hunk 1 above)
Comment (simahawk) [reply website_multi/menuitems.xml:10]: ouch, my eyes got tricked by multi/menu switch... nevermind! :sweat_smile:

File: website_multi/models/website.py (hunk 2 above)
Comment (tarteo) [reply website_multi/models/website.py]: For now the switcher only uses this field. But I have another module website_multi_blog (no PR yet) that uses this field to generate the RSS for multiple website properly. I can still use the web.base.url property as default in that module.

**Review by yajo — COMMENTED** [review]
I'm just wondering if this duplicates some features from https://www.odoo.com/apps/modules/11.0/website_multi_theme/ :thinking: 

Did you know about it?

**Comment by tarteo** [comment]
@Yajo Thanks for the reaction. Yes I know about the module and some are duplicate. If you want I can make some small changes to website_multi_theme in this PR e.g. the menu item 'Websites' is duplicate. website_multi_theme will then depend on this module. What do you think? 
We can later extend the multi website functionality by adding options for the footer (automatic / custom) etc to be website specific. In a live stream of Odoo I heard they are also working on multi website functionalities.

**Comment by yajo** [comment]
> If you want I can make some small changes to website_multi_theme in this PR e.g. the menu item 'Websites' is duplicate. website_multi_theme will then depend on this module. What do you think?

OK, it would be nice. @yelizariev will be interested in this too.



> We can later extend the multi website functionality by adding options for the footer (automatic / custom) etc to be website specific.

Right now, the website-specific views is, basically, the main feature for `website_multi_theme`. Just for you to know it. :blush: 

> In a live stream of Odoo I heard they are also working on multi website functionalities.

True, but I wouldn't count on it for odoo 12.

Thanks!

**Comment by yelizariev** [comment]
``website_multi_theme`` is useful even when you use the same theme for all websites. I.e. is not just "multi-theme" feature. 

> If you want I can make some small changes to website_multi_theme in this PR e.g. the menu item 'Websites' is duplicate. website_multi_theme will then depend on this module. What do you think?

I'd prefer to see all updates directly in ``website_multi_theme``. I don't think that multi-website is complete without multi-theme (multi-view) features. While making them separate force us to test ``website_multi_theme`` separately on every update of suggested ``website_multi`` module.

@tarteo be aware of our multi-website modules to don't reinvent the same features 
https://github.com/it-projects-llc/website-addons/tree/10.0/website_multi_company
https://github.com/it-projects-llc/website-addons/tree/10.0/website_multi_company_blog
https://github.com/it-projects-llc/website-addons/tree/10.0/website_multi_company_portal
https://github.com/it-projects-llc/website-addons/tree/10.0/website_multi_company_sale

**Comment by tarteo** [comment]
@Yajo Thanks! didn't know the website_multi_theme module has such great features :astonished:  .@yelizariev What do you suggest? I can put the functionalities that I've added in this module into the website_multi_theme module that includes:

- Improve the list view of websites with extra fields (website_ids)
- Easy switch between websites
- Change the website of pages in the page properties dialog

The improvements are mostly nice controls for the user. This module was implemented at a customer that wanted a website with multiple domains. You can see the result on www.ubports.com and www.ubuntu-touch.io. it's the same database using the same theme but with different content.

**Comment by tarteo** [comment]
@yelizariev I've also created a module website_multi_blog which takes care of the sitemap.xml and proper base_url. I can suggest this with a PR extending the module website_multi_company_blog. Thank you a lot for the references :+1:

**Comment by tarteo** [comment]
https://github.com/OCA/website/pull/488

**Pull request closed without merging**